intents/oversights/top_down_error.py

```diff
@@ -0,0 +1,247 @@
+            new_required_columns.append(metric)
+            new_cropped_table = aspects.crop_other_columns(table, 
+                                                           new_required_columns)
+
+            # result table after adding the new column in the grouping list.
+            new_result_table = aspects.group_by(new_cropped_table,
+                                                new_grouping_columns,
+                                                summary_operator)
+
+            # it will return the debiasing suggestion after comparing the
+            # initial result table and new result table.
+            new_suggestion =  _check_top_down_error(initial_result_table,
+                                                      new_result_table, 
+                                                      column, slice1,
+                                                      dimensions_len)
+            if new_suggestion != None:
+                suggestions.append(new_suggestion)
+    return suggestions
+
+def _check_top_down_error(initial_result_table, new_result_table, new_added_column,
+                                                            slice1, dimensions_len):
+    """
+    Args:
+        initial_result_table: Type-pandas.dataframe
+            It has the contents of the csv file
+        new_result_table: Type-pandas.dataframe
+            It has the contents of the csv file
+        new_added_column: Type-string
+            The new column added in the initial grouping column.
+        slice1: Type-string
+            the first slice by which we do comparision.
+        dimensions_len: Type-integer
+            size of the initial gfrouping dimensions.
+
+    Returns:
+        return a dictionary where each dictionary represent a debiasing 
+               suggestion according to the new column.
+    """
+    initial_num_rows = initial_result_table.shape[0]
+    new_num_rows = new_result_table.shape[0]
+
+    # changing dataframe to list of list to do next set of operation
+    initial_table_matrix = initial_result_table.values.tolist()
+    new_table_matrix = new_result_table.values.tolist()
+
+    initial_row_i = 0
+    new_row_i = 0
+
+    suggestion_row_list = []
+
+    # dominant percentage is the percentage of pairs where 
+    # (first value - second value) is positive.
+
+    while initial_row_i < initial_num_rows:
+        if initial_row_i == initial_num_rows - 1 or initial_table_matrix[initial_row_i][:dimensions_len] != initial_table_matrix[initial_row_i + 1][:dimensions_len]:
+            while new_row_i < new_num_rows and initial_table_matrix[initial_row_i][:dimensions_len] == new_table_matrix[new_row_i][:dimensions_len]:
+                new_row_i = new_row_i + 1
+        else:
+            new_max_corelation = 0
+            while new_row_i < new_num_rows and initial_table_matrix[initial_row_i][:dimensions_len] == new_table_matrix[new_row_i][:dimensions_len]:
```

Comment:
Use \ to break lines.

Reply:
Example - https://github.com/googleinterns/debaised-analysis/blob/master/date_detection/date_column_type_detection.py#L41

Reply:
will do thanks

Reply:
@11chankun  your forked is 1 commit behind the googleinterns:master , just update it with googleinterns. 
Thanks

Reply:
thanks done pull